misc/server/Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM rust:1.66-bullseye as builder
+WORKDIR /usr/src/rust-libp2p-server
+
+COPY . .
+RUN cargo install --locked --path .
```

Comment:
Let's use caching here to make local builds faster.

Reply:
With caching you are referring to mounting `.target/`, right? Done.